src/parser/monk/brewmaster/modules/spells/GiftOfTheOx.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber } from 'common/format';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import EventFilter from 'parser/core/EventFilter';
+import StatisticBox from 'interface/others/StatisticBox';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import { calculatePrimaryStat } from 'common/stats';
+
+import { BASE_AGI, GIFT_OF_THE_OX_SPELLS } from '../../constants';
+
+const WDPS_BASE_ILVL = 310;
+const WDPS_310_AGI_POLEARM = 122.8;
+
+/**
+ * Gift of the Ox
+ *
+ * Generated healing spheres when struck, which heal for 1.5x AP when
+ * consumed by walking over, expiration, overcapping, or casting
+ * Expel Harm.
+ *
+ * See peak for a breakdown of how it works and all its quirks:
+ * https://www.peakofserenity.com/2018/10/06/gift-of-the-ox/
+ */
+export default class GiftOfTheOx extends Analyzer {
+  static dependencies = {
+    stats: StatTracker,
+  }
+
+  totalHealing = 0;
+  agiBonusHealing = 0;
+  wdpsBonusHealing = 0;
+  _baseAgiHealing = 0;
+  masteryBonusHealing = 0;
+  _wdps = 0;
+
+  orbsGenerated = 0;
+  orbsConsumed = 0;
+
+  expelHarmCasts = 0;
+  expelHarmOrbsConsumed = 0;
+  expelHarmOverhealing = 0;
+
+  _lastEHTimestamp = null;
+
+  constructor(...args) {
+    super(...args);
+    this.addEventListener(new EventFilter('tick').by(SELECTED_PLAYER).spell(GIFT_OF_THE_OX_SPELLS), this._orbGenerated);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EXPEL_HARM), this._expelCast);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(GIFT_OF_THE_OX_SPELLS), this._gotoxHeal);
+
+    this._wdps = calculatePrimaryStat(WDPS_BASE_ILVL, WDPS_310_AGI_POLEARM, this.selectedCombatant.mainHand.itemLevel);
+  }
+
+  _orbGenerated(event) {
+    this.orbsGenerated += 1;
+  }
+
+  _expelCast(event) {
+    this.expelHarmCasts += 1;
+    this._lastEHTimestamp = event.timestamp;
+  }
+
+  _gotoxHeal(event) {
+    this.orbsConsumed += 1;
+    const amount = event.amount + (event.absorbed || 0);
+    this.totalHealing += amount;
+
+    // so the formula for the healing is
+    //
+    // Heal = 1.5 * (6 * WDPS + BonusAgi + BaseAgi) * Mastery * Vers
+    //
+    // With BaseAgi known, we get:
+    //
+    // BonusHeal = 1.5 * (6 * WDPS + BonusAgi + BaseAgi) * Mastery * Vers - 1.5 * (6 * WDPS + BaseAgi) * Mastery * Vers
+    //           = Heal * (1 - (6 WDPS + BaseAgi) / (6 WDPS + BonusAgi + BaseAgi))
+    //           = Heal * (BonusAgi / (6 WDPS + BonusAgi + BaseAgi))
+    //
+    // and similar for bonus WDPS healing and base agi healing
+    const denom = (6 * this._wdps + this.stats.currentAgilityRating);
```

Comment:
I don't really understand the math here, why and how, so I'm just gonna trust you on this 😅  If it's wrong, it's wrong just for BMs 🤷‍♀️